add-chain/check-rollup-config.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"os"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/urfave/cli/v2"
+)
+
+var CheckRollupConfigCmd = cli.Command{
+	Name:  "check-rollup-config",
+	Flags: []cli.Flag{RollupConfigFlag},
+	Usage: "Sanity check rollup config (rollup.json) is reproducible",
+	Action: func(ctx *cli.Context) error {
+		// Ensure rollup config generated by op-node right now is equivalent to pre-existing
+		// one used to create superchain config files. i.e. compare the following:
+		//  1. op-node(superchain/configs/<l1-chain>/<l2-chain>.yaml) --> rollup.Config
+		//  2. monorepo/op-node/rollup.json Unmarshal --> rollup.Config
+		rollupConfigPath := RollupConfigFlag.Get(ctx)
+
+		fmt.Printf("Attempting to read from %s\n", rollupConfigPath)
+		file, err := os.ReadFile(rollupConfigPath)
+		if err != nil {
+			return fmt.Errorf("failed to read from local rollup.json config file: %w", err)
+		}
+		var localRollupConfig *rollup.Config
+		if err = json.Unmarshal(file, &localRollupConfig); err != nil {
+			return fmt.Errorf("failed to unmarshal local rollup.json into rollup.Config struct: %w", err)
+		}
+
+		opNodeRollupConfig, err := rollup.LoadOPStackRollupConfig(localRollupConfig.L2ChainID.Uint64())
+		if err != nil {
+			return fmt.Errorf("op-node failed to generate rollup config: %w", err)
+		}
+		opts := cmp.Options{
+			cmpopts.IgnoreUnexported(big.Int{}),
+		}
+
+		// Exclude this field from check since it gets populated by op-node after rollup.json is created
+		localRollupConfig.ProtocolVersionsAddress = common.Address{}
+		opNodeRollupConfig.ProtocolVersionsAddress = common.Address{}
```

Comment:
I'm not entirely satisfied with this but even with this exception we are getting a lot of protection from other things not matching. We should create an issue and circle back to this.

If we wanted to follow the pattern with hardfork activation times, we would:
* unexport the ProtocolVersions field from SuperchainConfig type. 
* add it to the ChainConfig type and set it to the value from `rollup.json` during the `add-chain` flow
* have logic to conditionally hydrate the chain config type with the value from the superchain config type, depending on whether `superchain_time` is `nil` or not `nil` and whether ProtocolVersions is empty or not.
* have op-node read from the chain config type instead of the superchain config type (it now can't anyway because it is private). 

This way, the discrepancy in this new tool you have written would be resolved by new chains not having the `superchain_time` set and therefore op-node would not be hydrating a protocol versions address from the superchain.yaml file and it should match.

Reply:
Created a new issue for this follow-up effort: https://github.com/ethereum-optimism/superchain-registry/issues/334